source/adam.cpp

```diff
@@ -345,6 +346,27 @@ class sheet_t::implementation_t : boost::noncopyable {
     priority_t name_to_priority(name_t name) const;
     void flow(cell_bits_t& priority_accessed);
 
+    /// Returns the output cell with the given name.
+    /// \pre the output cell must exist.
+    cell_t& output_cell(const name_t& name) {
+        auto p = output_index_m.find(name);
+        assert(p != output_index_m.end() && "output cell not found");
+        return *p;
+    }
+    const cell_t& output_cell(const name_t& name) const {
+        auto p = output_index_m.find(name);
+        assert(p != output_index_m.end() && "output cell not found");
+        return *p;
+    }
+
+    /// Returns `true` if any output cells in the relation are resolved, false otherwise.
```

Comment:
Some people think iff is too obscure, but this is an option

```suggestion
    /// Returns `true` iff any output cells in the relation are resolved.
```
Some people like:
```suggestion
    /// Returns whether any output cells in the relation are resolved.
```

I find that really awkward personally but I can see the argument.

If you are in the context of a type representing a relation, you can leave out “in the relation.”

Reply:
My preference was iff (I had that initially, but I know people who would object, so I changed it).  I'll go with "whether" -